packages/rich-list-core/src/RichListCore.ts

```diff
@@ -0,0 +1,179 @@
+import { ApiClient, BigNumber, blockchain as defid } from '@defichain/jellyfish-api-core'
+import { QueueClient } from './lib/Queue'
+import { PersistentStorage } from './lib/PersistentStorage'
+import { AddressParser } from './controller/AddressParser'
+import { NetworkName } from '@defichain/jellyfish-network'
+import { RichListItem } from '@defichain/rich-list-api-client'
+import { AccountAmount } from 'packages/jellyfish-api-core/src/category/account'
+import { ActiveAddressAccountAmount } from './controller/AddressParser/ActiveAddressAccountAmount'
+import { PersistentLinkedList } from './lib/PersistentLinkedList'
+
+export class RichListCore {
+  CATCHING_UP = false
+  RICH_LIST_LENGTH = 1000
+  readonly addressParser: AddressParser
+
+  constructor (
+    private readonly network: NetworkName,
+    private readonly apiClient: ApiClient,
+    private readonly existingRichList: PersistentStorage<number, RichListItem[]>,
+    private readonly queueClient: QueueClient<string>,
+    private readonly crawledBlocks: PersistentLinkedList<BlockDroppedOutAddresses>
+  ) {
+    this.addressParser = new AddressParser(apiClient, network)
+  }
+
+  setRichListLength (length: number): void {
+    this.RICH_LIST_LENGTH = length
+  }
+
+  resume (): void {
+    if (this.CATCHING_UP) {
+      return
+    }
+    this.CATCHING_UP = true
+    void this._catchUp()
+  }
+
+  async _catchUp (): Promise<void> {
```

Comment:
Are `_` prefixed methods meant to represent `private`, in which case perhaps we should use the `private` class assignment for these methods to be more explicit?

Reply:
Actually looking through the file I see some `_` with preceding `private` assignment. What does the `_` represent in this class?

Reply:
All they `_` prefixed are meant to be private. But during the implementation I keep changing my mind for exposing them for functional test, ended up the inconsistency here. 

Will make them private first, more likely to have e2e instead of actually testing each single function.